exchanges/poloniex/poloniex.go

```diff
@@ -16,920 +16,1162 @@ import (
-	result := MoveOrderResponse{}
-	values := url.Values{}
-
-	if orderID == 0 {
-		return result, errors.New("orderID cannot be zero")
+// MaximumBuySellAmount get maximum and available buy/sell amount for a given symbol.
+func (e *Exchange) MaximumBuySellAmount(ctx context.Context, symbol currency.Pair) (*MaxBuySellAmount, error) {
+	if symbol.IsEmpty() {
+		return nil, currency.ErrCurrencyPairEmpty
 	}
+	params := url.Values{}
+	params.Set("symbol", symbol.String())
+	var resp *MaxBuySellAmount
+	return resp, e.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, authNonResourceIntensiveEPL, http.MethodGet, "/margin/maxSize", params, nil, &resp)
+}
 
-	if rate == 0 {
-		return result, errors.New("rate cannot be zero")
+// PlaceOrder places an order
+func (e *Exchange) PlaceOrder(ctx context.Context, arg *PlaceOrderRequest) (*PlaceOrderResponse, error) {
+	if arg.Symbol.IsEmpty() {
+		return nil, currency.ErrCurrencyPairEmpty
 	}
-
-	values.Set("orderNumber", strconv.FormatInt(orderID, 10))
-	values.Set("rate", strconv.FormatFloat(rate, 'f', -1, 64))
-
-	if postOnly {
-		values.Set("postOnly", "true")
+	if arg.Side == "" {
+		return nil, order.ErrSideIsInvalid
 	}
-
-	if immediateOrCancel {
-		values.Set("immediateOrCancel", "true")
+	if arg.Amount <= 0 {
+		return nil, limits.ErrAmountBelowMin
 	}
-
-	if amount != 0 {
-		values.Set("amount", strconv.FormatFloat(amount, 'f', -1, 64))
+	var resp *PlaceOrderResponse
+	if err := e.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, authResourceIntensiveEPL, http.MethodPost, "/orders", nil, arg, &resp); err != nil {
+		return nil, err
 	}
+	if resp == nil {
+		return nil, common.ErrNoResponse
+	} else if resp.Code != 0 && resp.Code != 200 {
+		return resp, fmt.Errorf("%w: code: %d message: %s", common.ErrNoResponse, resp.Code, resp.Message)
+	}
+	return resp, nil
+}
 
-	err := e.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, http.MethodPost,
-		poloniexOrderMove,
-		values,
-		&result)
+// PlaceBatchOrders places a batch of orders
+func (e *Exchange) PlaceBatchOrders(ctx context.Context, args []PlaceOrderRequest) ([]*PlaceBatchOrderItem, error) {
+	if len(args) == 0 {
+		return nil, common.ErrNilPointer
+	}
+	for x := range args {
+		if args[x].Symbol.IsEmpty() {
+			return nil, currency.ErrCurrencyPairEmpty
+		}
+		if args[x].Side == "" {
+			return nil, order.ErrSideIsInvalid
+		}
+		if args[x].Amount <= 0 {
+			return nil, limits.ErrAmountBelowMin
+		}
+	}
+	var resp []*PlaceBatchOrderItem
+	err := e.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, authResourceIntensiveEPL, http.MethodPost, "/orders/batch", nil, args, &resp)
 	if err != nil {
-		return result, err
+		return nil, err
 	}
-
-	if result.Success != 1 {
-		return result, errors.New(result.Error)
+	for _, r := range resp {
+		if r.Code != 0 && r.Code != 200 {
+			err = common.AppendError(err, fmt.Errorf("%w: code: %d message: %s", common.ErrNoResponse, r.Code, r.Message))
```

Comment:
🚧 Change request:
I don't think that `common.ErrNoResponse` is correct here.
Can we not tell the user anything useful about which order just failed?
ClientOrderID and index would be a good idea.